ast/compile.go

```diff
@@ -2852,49 +3029,56 @@ type TreeNode struct {
 	Hide     bool
 }
 
+func (n *TreeNode) String() string {
+	return fmt.Sprintf("<TreeNode key:%v values:%v sorted:%v hide:%v>", n.Key, n.Values, n.Sorted, n.Hide)
+}
+
 // NewRuleTree returns a new TreeNode that represents the root
 // of the rule tree populated with the given rules.
 func NewRuleTree(mtree *ModuleTreeNode) *TreeNode {
```

Comment:
The RuleTree creation was basically rewritten. I hadn't been able to reconcile the previous code with the new Ref rules. The new implementation does less things implicitly:

1. If there's a system module, it's set to hidden. For this, it now knows the path of the system nodes -- previously, only the module tree knew that (hardcoded) information; now it's also inserted here.
2. it sorts after creation.